backend/src/query-validation.pipe.ts

```diff
@@ -0,0 +1,27 @@
+import { ArgumentMetadata, Injectable, PipeTransform } from '@nestjs/common';
+import { Type, plainToInstance } from 'class-transformer';
+import { IsInt, IsString } from 'class-validator';
+
+export class QueryDto {
+  @Type(() => Number)
```

Comment:
What is @Type achieving?

Reply:
We are telling class transformer that this should be a number.